### PR TITLE
ceph: use next ceph v16.2.6 pacific version

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by travisn — CHANGES_REQUESTED** [review]
LGTM, but adding changes requested until the cephfs issue is resolved

**Comment by leseb** [comment]
> LGTM, but adding changes requested until the cephfs issue is resolved

Fixed in the third commit.

**Review by travisn — COMMENTED** [review]
This new approach also works for pre-16.2.6, right? I guess we can see with the integration tests...

**Comment by leseb** [comment]
> This new approach also works for pre-16.2.6, right? I guess we can see with the integration tests...

Yes it should, @batrick suggested that we should have had it like this for some time actually.
However, the integration runs on 16.2.6 now so the CI won't validate this.

File: cluster/examples/kubernetes/ceph/cluster.yaml [hunk 1]
```diff
@@ -85,15 +85,15 @@ spec:
...
-      # The selector keys are required to be `public` and `cluster`.
-      # Based on the configuration, the operator will do the following:
-      #   1. if only the `public` selector key is specified both public_network and cluster_network Ceph settings will listen on that interface
-      #   2. if both `public` and `cluster` selector keys are specified the first one will point to 'public_network' flag and the second one to 'cluster_network'
-      #
-      # In order to work, each selector value must match a NetworkAttachmentDefinition object in Multus
-      #
-      #public: public-conf --> NetworkAttachmentDefinition object name in Multus
-      #cluster: cluster-conf --> NetworkAttachmentDefinition object name in Multus
+    # The selector keys are required to be `public` and `cluster`.
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster.yaml:88]: Can these whitespace changes be reverted? Or back to the question of having a separate PR for whitespace consistency?

File: pkg/operator/ceph/file/filesystem.go [hunk 2]
```diff
@@ -51,37 +49,31 @@ func createFilesystem(
 	ownerInfo *k8sutil.OwnerInfo,
 	dataDirHostPath string,
 ) error {
+	logger.Infof("start running mdses for filesystem %q", fs.Name)
```
Comment (travisn) [inline pkg/operator/ceph/file/filesystem.go:52]: In the past, starting RGW first caused Ceph to create the pools. We don't have that same behavior with mds, right? Just wanted to confirm that Rook still owns pool creation.

File: pkg/operator/ceph/file/mds/mds.go [hunk 3]
```diff
@@ -249,7 +245,9 @@ func (c *Cluster) upgradeMDS() error {
 		return errors.Wrap(err, "failed to setting allow_standby_replay to false")
 	}
 
-	// In Pacific, standby-replay daemons are stopped automatically. Older versions of Ceph require us to stop these daemons manually.
+	// In Pacific, standby-replay daemons are stopped automatically. Older versions of Ceph require
+	// us to stop these daemons manually.
+	// TODO: so why don't we have a version check?
```
Comment (travisn) [inline pkg/operator/ceph/file/mds/mds.go:250]: This looks like it's always failing replay daemons whether it's pacific or not. @jshen28 Do we need to check for Pacific or was this comment outdated?

File: pkg/operator/ceph/file/filesystem.go (hunk 2 above)
Comment (leseb) [reply pkg/operator/ceph/file/filesystem.go:52]: Correct, pools are not auto-created if they don't exist AFAIK.

**Review by travisn — APPROVED** [review]

**Pull request merged**